app/src/main/java/au/id/neasbey/biblecast/util/BibleAPIRequest.java

```diff
@@ -0,0 +1,177 @@
+                requestUrl += "&version=" + URLEncoder.encode(bibleVersions, "UTF-8");
+
+            } catch (UnsupportedEncodingException e) {
+                Log.e(TAG, "URL encoding Error: " + e.getMessage());
+            }
+        }
+
+        return requestUrl;
+    }
+
+    public String performRequest() {
+        Log.d(TAG, "Request: " + requestUrl);
+
+        String resultToDisplay = "";
+
+        if(isRequestValid()) {
+            try {
+                HttpURLConnection urlConnection = createConnection(requestUrl, apiAuth);
+
+                checkServerResponseOk(urlConnection);
+                handleResponse(urlConnection);
+
+            } catch (Exception e) {
+                resultToDisplay = e.getMessage();
+            }
+        } else {
+            resultToDisplay = "Please enter a valid query";
+        }
+
+        return resultToDisplay;
+    }
+
+    public boolean isRequestValid() {
+        return !requestUrl.isEmpty();
+    }
+
+    public List<Spanned> getResultsList() {
+        return resultList;
+    }
+
+    private HttpURLConnection createConnection(String requestUrl, String apiAuth) throws IOException {
+        // Send HTTP Get request
+        URL url = new URL(requestUrl);
+        HttpURLConnection urlConnection = (HttpURLConnection) url.openConnection();
+
+        // Set API authentication
+        urlConnection.addRequestProperty("Authorization", generateHttpAuthentication(apiAuth.getBytes()));
+
+        return urlConnection;
+    }
+
+    private void handleResponse(HttpURLConnection urlConnection) throws Exception {
+        String responseJSON = getServerResponseJSON(urlConnection);
+
+        // Parse server response
+        BibleAPIResponseParser bibleAPIResponseParser = new BibleAPIResponseParser();
+        bibleAPIResponseParser.parseJSONToList(responseJSON, resultList);
+    }
+
+    /**
+     * Get server response JSON
+     */
+    private String getServerResponseJSON(HttpURLConnection urlConnection) throws Exception {
+
+        BufferedReader reader = null;
+        StringBuilder responseText = new StringBuilder();
+
+        try {
+            reader = new BufferedReader(new InputStreamReader(urlConnection.getInputStream()));
+            String line;
+
+            while ((line = reader.readLine()) != null) {
+                responseText.append(line);
+                responseText.append("\n");
+            }
+
+            Log.d(TAG, "Response: " + responseText.toString());
+        } catch (IOException ioe) {
+            throw new Exception(ioe.getMessage());
+        } finally {
+            try {
+                if (reader != null) {
+                    reader.close();
+                }
+            } catch (Exception ex) {
+                Log.e(TAG, "Failed to close input reader: " + ex.getMessage());
+            }
+        }
+
+        return responseText.toString();
+    }
+
+    private void checkServerResponseOk(HttpURLConnection urlConnection) throws Exception {
```

Comment:
I don't like throwing Exception ... it forces a catch Exception and will break exception handling.
One alternative would be to subclass Exception so that you know which Exceptions are created by your app.